### DISCUSSION TITLE
How to check if a user has an active Play Pass subscription?

### DISCUSSION THREAD

**Discussion by IShix-g** [KirillKuzyk/Google-Play-License-Check#6] [Q&A] [answered]
The game was released today. This is to support Google Play Pass. But it returns the result `license.status == LicenseStatus.LICENSED` even if it is not Google Play Pass.
`LICENSED` even though I have 3 devices and all of them are not subscribed to Google Play Pass.

So far, this plugin is the one that returns `LicenseStatus.LICENSED` if it is downloaded from Google Play. Is there any mistake?

- Currently, the ID used in the terminal is not registered in License Testing.
- License Testing has worked properly. All statuses have been verified. All statuses have been confirmed. 

It's part of my code.
```C#
isNowLoadingLicense = true;
GooglePlayLicense.Check(license =>
{
    AppManager.SetLicenseStatus(license.status == LicenseStatus.LICENSED);
    isNowLoadingLicense = false;
   // Game Startup
});
```

**Comment by KyryloKuzyk** [comment]
It seems like I wasn't completely correct in [this answer](https://github.com/KirillKuzyk/Google-Play-License-Check/discussions/5#discussioncomment-3270505) before.
The correct answer would be this:
```
If the status code is LICENSED, it can mean that:  
1. A user downloaded a free game on Google Play.
2. A user has purchased a paid game.
3. A user has an active Google Play Pass subscription.
```
So it's expected that a free game downloaded from Google Play has a LICENSED status.

**Comment by KyryloKuzyk** [comment, marked as answer]
Unfortunately, there is no way to check if a user has an active Play Pass subscription in case of a free game.

I checked how Google suggests integrating a free game with Google Play Pass. It follows like this:
1. You tell Google what IAP products should be granted automatically if a user has an active Play Pass subscription. These IAPs will be automatically unlocked for such a user.
2. All other IAP products become free for a user with an active Play Pass subscription.
3. When a user's Play Pass subscription becomes inactive, all previously granted IAP entitlements acquired through Play Pass are automatically revoked.

So in your case, you can register an IAP product, let's call it PREMIUM_CONTENT_PURCHASED. Then you should check if a user has a PREMIUM_CONTENT_PURCHASED entitlement. Unfortunately, the 'Google Play License Check' doesn't have the billing functionality and you have to use another plugin for that.

**Comment by IShix-g** [comment]
Thank you for your reply. Let's put the IAP product in perspective.
What I am looking for in the plugin is if it can determine if you are a Google Play Pass subscription user.

For example, if I show ads with IAP products, I should not show ads to Google Play Pass users. I need to be able to determine if they are Google Play Pass users and hide the ads. Does this mean that this plugin cannot handle this case?

**Comment by IShix-g** [comment]
> licenseValidityTimestamp should be equal to the 'DateTime.Now + 6 days'
> licenseValidityTimestamp should equal to the DateTime.Max.

Thank you for your reply. Could you please add the functionality for the plugin to work that way? I would like to be able to check it with flags in the following way.

```C#
GooglePlayLicense.Check(license =>
{
    if (license.status == LicenseStatus.LICENSED
        && license.IsSubscribedPlayPass)
    {
        // This user has a license for Google play pass
    }
});
```

**Reply by KyryloKuzyk** [reply]
Please disregard my suggestion to check the licenseValidityTimestamp. Unfortunately, it's not possible to determine if a user has an active Play Pass subscription in case of a free game. 
Google suggests using IAP for such cases, as described here:
https://github.com/KirillKuzyk/Google-Play-License-Check/discussions/6#discussioncomment-3314455

Please refer to the Play Pass documentation for more info on the implementation.

**Comment by IShix-g** [comment]
I knew that Google Play Pass could be achieved simply by implementing IAP. I was looking for a plugin since it would not accomplish my goal. This plugin does nothing for Google Play Pass. I suggest you fix the plugin page.

In the future I would like to use this plugin to check if I have downloaded from Google Play. Very useful. Thanks for the explanation.

**Reply by KyryloKuzyk** [reply]
> This plugin does nothing for Google Play Pass. I suggest you fix the plugin page.

Yes, I agree. I'll update the plugin's description to reflect that the plugin is not suitable for free Play Pass games.
The plugin is only useful for paid Play Pass games. 
Thank you!
